src/libsystemd/sd-journal/journal-authenticate.c

```diff
@@ -410,9 +413,10 @@ int journal_file_hmac_setup(JournalFile *f) {
         if (!JOURNAL_HEADER_SEALED(f->header))
                 return 0;
 
-        initialize_libgcrypt(true);
+        if (initialize_libgcrypt(true) < 0)
+                return -EOPNOTSUPP;
```

Comment:
Should return the actual error

Reply:
I'm not sure - this is propagated from a public sd_journal API, and the call below that further initializes gcrypt also returns EOPNOTSUPP on failure, so I thought it would be best to keep the error consistent, for external users

Reply:
Realistically the only error `initialize_libgcrypt()` is going to return is `EOPNOTSUPP`, but it will also return a custom error if a symbol is missing from the library. I think we should propagate that to help users debug problems.